compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1303,7 +1299,9 @@ private boolean validateTableConstructorExpr(BLangTableConstructorExpr tableCons
     }
 
     public void validateMapConstraintTable(BLangTableConstructorExpr tableConstructorExpr, BType expType) {
-        if (((BTableType) expType).fieldNameList != null || ((BTableType) expType).keyTypeConstraint != null) {
+        if (expType != null && (((BTableType) expType).fieldNameList != null ||
+                ((BTableType) expType).keyTypeConstraint != null) &&
+                !expType.tsymbol.owner.getFlags().contains(Flag.LANG_LIB)) {
```

Comment:
Why are we filtering out the lang_lib. Is there a way to handle the map constraint in a generic manner?

Reply:
In lang.table there are some places that have specified a `key-type-constraint` for table constrained with map [1], which results in `table with constraint of type map cannot have key specifier or key type constraint` compilation error. Therefore it's handled here with the langlib check here.
[1] https://github.com/ballerina-platform/ballerina-lang/blob/master/langlib/lang.table/src/main/ballerina/table.bal#L77